src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -152,4 +184,87 @@ export function convertScatterDataToDownloadData(patientData: any[]): any[] {
     })
 
     return downloadData;
+}
+
+/**
+ * Down sampling based on the hypotenuse difference.
+ * The cross dot is guaranteed to be drawn when hidden dots are taking into account.
+ * @param {DownSampling} opts
+ * @returns {ScatterData[]}
+ */
+export function downSampling(data: ScatterData[], opts: DownSamplingOpts): ScatterData[] {
+    let x = data.map(item => item.x);
+    let y = data.map(item => item.y);
+
+    let xMax = _.max(x) || 0;
+    let xMin = _.min(x) || 0;
+
+    let yMax = _.max(y) || 0;
+    let yMin = _.min(y) || 0;
+
+    if (opts.xDenominator <= 0 || opts.yDenominator <= 0)
+        return data;
+
+    let timeThreshold = (xMax - xMin) / opts.xDenominator;
+    let survivalRateThreshold = (yMax - yMin) / opts.yDenominator;
+    let averageThreshold = Math.hypot(timeThreshold, survivalRateThreshold);
```

Comment:
I just checked and looks like Math.hypot is not supported in IE11.  May see if jStat has it.  Or install a polyfill.